### DISCUSSION TITLE
Sentry Native Support without NativeAOT

### DISCUSSION THREAD

**Discussion by DominicMaas** [getsentry/sentry-dotnet#3478] [Q&A]
Hi, First, amazing work with the SDK and product! It's a dream to use compared to AppCenter.

I've got a WinUI application that talks to a shared c++ library that's not compiled using NativeAOT (not supported on WinUI at the moment). 

I'm wondering what's the best way forward with catching errors in the native code? Managed code is already implemented and works beautifully.

I was looking at integrating sentry native into the library, but I noticed that some work was done to allow catching of native crashes for NativeAOT scenarios (but this is gated to aot apps from what I can understand).

Would the native integration in this SDK suffice if i could enable it for non-aot scenarios, or shall I continue with integrating sentry into the native library?

**Comment by bitsandfoxes** [comment]
Hey @DominicMaas, glad you're getting some value out of the SDK!

That is actually really tricky and I don't know how to answer that off the top of my head. Looking at our `.csproj` the native target has no requirement on AOT:
https://github.com/getsentry/sentry-dotnet/blob/e68c8c7e2d165715077e12578679be15192e3537/src/Sentry/Sentry.csproj#L24-L25

**Comment by DominicMaas** [comment]
Hi!

Looking at https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/Platforms/Native/SentryNative.cs it seems that there is an explicit check on if the app is running under NativeAOT. (I assume for good reason).

Yeah it's a little tricky! Our macOS app (using the same core c++ library) has native integration, so went on a expedition to see if I can integrate the same into our Windows app.

**Reply by bitsandfoxes** [reply]
So it looks like right now there is no proper way for you to work around this limitation. `sentry-native` got added to support AOT - which is why there's that check in there. We never got around to check whether this now works out-of-the-box without AOT but that'd be a valuable thing to figure out!

**Comment by DominicMaas** [comment]
Just came across these issues that relate to my situation I believe:

#3244

#2076

**Comment by DominicMaas** [comment]
Following up on this, our app is now fully switched over to NativeAOT, and WinUI + a lot of the popular libraries are AOT compatible now. So I imagine this is quite a niche use case.

**Reply by DominicMaas** [reply]
oh, although NativeAOT is only supported on Android/iOS?
